cmd/src/extsvc_add.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/mattn/go-isatty"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/src-cli/internal/api"
+)
+
+func init() {
+	usage := `
+  Examples:
+
+  Add an external service configuration on the Sourcegraph instance:
+
+  $ cat new-config.json | src extsvc add
+  $ src extsvc add -name 'My GitHub connection' new-config.json
+  `
+
+	flagSet := flag.NewFlagSet("add", flag.ExitOnError)
+	usageFunc := func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src extsvc %s':\n", flagSet.Name())
+		flagSet.PrintDefaults()
+		fmt.Println(usage)
+	}
+	var (
+		nameFlag = flagSet.String("name", "", "exact name of the external service to add")
+		kindFlag = flagSet.String("kind", "", "kind of the external service to add")
+		apiFlags = api.NewFlags(flagSet)
+	)
+
+	handler := func(args []string) (err error) {
+		ctx := context.Background()
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		if *nameFlag == "" {
+			return errors.New("-name must be provided")
+		}
+
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+		if *nameFlag != "" {
+			_, err := lookupExternalService(ctx, client, "", *nameFlag)
+			if err != errServiceNotFound {
+				return errors.New("service already exists")
+			}
+		}
```

Comment:
Do you need to do that lookup or can't you be optimistic and have the server fail if there's a problem with a duplicate name?

Reply:
I naively assumed we didn't want duplicate and this wasn't supported. Fixed as we actually support this.